debugger/src/debugger.rs

```diff
@@ -1072,18 +1076,158 @@ impl Debugger {
+                        page_size_done: usize,
+                        sector_size_done: usize,
+                        fill_size_done: usize,
+                    }
 
-                let flash_result = download_file_with_options(
-                    &mut session_data.session,
-                    path_to_elf,
-                    Format::Elf,
-                    download_options,
-                );
+                    let flash_progress = Rc::new(RefCell::new(ProgressState {
+                        total_page_size: 0,
+                        total_sector_size: 0,
+                        total_fill_size: 0,
+                        page_size_done: 0,
+                        sector_size_done: 0,
+                        fill_size_done: 0,
+                    }));
+
+                    let flash_progress = if let Some(id) = progress_id {
+                        FlashProgress::new(move |event| {
+                            let mut flash_progress = flash_progress.borrow_mut();
+                            let mut debug_adapter = rc_debug_adapter_clone.borrow_mut();
+                            match event {
+                                probe_rs::flashing::ProgressEvent::Initialized { flash_layout } => {
+                                    flash_progress.total_page_size = flash_layout
+                                        .pages()
+                                        .iter()
+                                        .map(|s| s.size() as usize)
+                                        .sum();
+
+                                    flash_progress.total_sector_size = flash_layout
+                                        .sectors()
+                                        .iter()
+                                        .map(|s| s.size() as usize)
+                                        .sum();
+
+                                    flash_progress.total_fill_size = flash_layout
+                                        .fills()
+                                        .iter()
+                                        .map(|s| s.size() as usize)
+                                        .sum();
+                                }
+                                probe_rs::flashing::ProgressEvent::StartedFilling => {
+                                    debug_adapter
+                                        .update_progress(0.0, Some("Reading Old Pages ..."), id)
+                                        .ok();
+                                }
+                                probe_rs::flashing::ProgressEvent::PageFilled { size, .. } => {
+                                    flash_progress.fill_size_done += size as usize;
+                                    debug_adapter
+                                        .update_progress(
+                                            flash_progress.total_fill_size as f64
+                                                / flash_progress.fill_size_done as f64,
```

Comment:
This should be the other way round to calculate the percentage:
```suggestion
                                            flash_progress.fill_size_done as f64
                                                / flash_progress.total_fill_size as f64,
```

Reply:
Oh that's why it always showed as complete ... silly me ... thanks!